catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergErrorMapper.java

```diff
@@ -89,6 +96,30 @@ public Response toResponse(Throwable ex, IcebergEntityKind kind) {
     return Response.status(code == null ? 500 : code).entity(body).build();
   }
 
+  private static String message(BackendErrorStatus status, Throwable ex) {
+    return String.format("%s (due to: %s)", ex.getMessage(), status.cause().toString());
+  }
+
+  private IcebergErrorResponse mapStorageFailure(BackendErrorStatus status, Throwable ex) {
+    // Log full stack trace on the server side for troubleshooting
+    LOGGER.info("Propagating storage failure to client: {}", status, ex);
+
+    switch (status.statusCode()) {
+      case THROTTLED:
+        return errorResponse(429, "RuntimeException", message(status, ex), ex);
+      case UNAUTHORIZED:
+        return errorResponse(401, "NotAuthorizedException", message(status, ex), ex);
+      case FORBIDDEN:
+        return errorResponse(403, "ForbiddenException", message(status, ex), ex);
+      case NOT_FOUND:
+        // Convert storage-side "not found" into `IllegalArgumentException`.
+        // In most cases this results from bad locations in Iceberg metadata.
+        return errorResponse(400, "IllegalArgumentException", message(status, ex), ex);
+      default:
+        return null;
```

Comment:
```suggestion
      case UNKNOWN:
        return null;
      default:
        throw new IllegalStateException("Unexpected status: " + status);
```

Reply:
Throwing from an exception mapper will probably result in a cryptic client-side error :thinking: 

I'll change the `default` to 500 mentioning the unknown enum value in the message.

Reply:
updated